### PR TITLE
fix: redirect without appending extra chars

### LINKED ISSUE #1204: Unwanted %2F appended to url when typed in the url bar
### Version
nuxt-i18n: `^6.22.3`
nuxt: `^2.9.2`

 - [x] universal
 - [ ] spa

### Nuxt-i18n configuration
```javascript
import en from "./locales/en.json"
import fr from "./locales/fr.json"

// ...
export default {
  vueI18n: {
    locale: "en",
    fallbackLocale: "en",
    messages: { en, fr },
    locales: [
      {
        code: "en",
        name: "English",
        iso: "en-US",
      },
      {
        code: "fr",
        name: "Français",
        iso: "fr-FR",
      },
    ],
    seo: true,
    defaultLocale: "en",
    strategy: "prefix",
    lazy: false,
    langDir: "~/locales/",
    vuex: {
      moduleName: "i18n",
      syncLocale: true,
    },
    detectBrowserLanguage: {
      useCookie: true,
      alwaysRedirect: true,
    },
  },
}
```

### Steps to reproduce
We'll consider that in this example that the nuxt app supports `en` as it's language
For this given url : `/foo/bar/?q=value`
We would expect the app to resolve this into `/en/foo/bar/?q=value` when typed in the url bar
Sadly in my case the module just resolves this to `/en/foo/bar/?q=value%2F`

### What is Expected?
The url to remain without any `%2F` occurences

### What is actually happening?
The url should stay as is when first typed in

### PR DESCRIPTION
**Pull request by divine**
Closes #1204

We might simplify it when https://github.com/unjs/ufo/pull/24 will be merged.

@rchl what do you think?

### CODE REVIEW

File: src/templates/plugin.routing.js
Comment (rchl) [inline src/templates/plugin.routing.js]: Looks good but I would rather delegate joining back the path to ufo's `stringifyParsedURL`.

Comment (divine) [reply src/templates/plugin.routing.js]: Do you mean this?

```javascript
const parsedPath = parsePath(localizedRoute.path)
parsedPath.pathname = nuxtOptions.trailingSlash ? withTrailingSlash(parsedPath.pathname) : withoutTrailingSlash(parsedPath.pathname)
localizedRoute.path = stringifyParsedURL(parsedPath)
```

It's adding extra `?` and basically breaks, looks like a bug https://github.com/unjs/ufo/blob/2a21bc6b821eafd2284cacdf1f99a4bd133c1fd7/src/parse.ts#L68

Thanks!

Comment (rchl) [reply src/templates/plugin.routing.js]: I don't have much confidence in ufo at this point...

Comment (divine) [reply src/templates/plugin.routing.js]: I have created a new issue https://github.com/unjs/ufo/issues/28

Maybe I should take a look at another library that might do better.

Thanks!

Comment (divine) [reply src/templates/plugin.routing.js]: We don't have much choice as there is no alternative.

I have created a fix, let's see when this will be merged, I'll update this PR  with your suggested changes.

Thanks!

**Comment by divine** [comment]
https://github.com/unjs/ufo/pull/24 has been merged.

Should be now better.

Thanks!

File: src/helpers/routes.js
Comment (rchl) [inline src/helpers/routes.js]: The `(isChildWithRelativePath ? '' : '/')` could  stay as before, I think. Even if the new logic is equivalent (hard to say for sure), the old one seems easier to read.

Also shouldn't `trailingSlash ? withTrailingSlash(path, true) : withoutTrailingSlash(path, true)` be wrapped in parenthesis? Otherwise the logic might not work exactly as intended.

File: src/templates/plugin.routing.js
Comment (rchl) [inline src/templates/plugin.routing.js:91]: What about the case of `/` path and `trailingSlash = false`? Does this end up with empty string or `/` (as it should)?

File: test/browser.test.js
Comment (rchl) [inline test/browser.test.js]: Those tests pass even without the fix so probably not testing the right thing?

**Review by rchl — CHANGES_REQUESTED** [review]

File: src/templates/plugin.routing.js
Comment (divine) [reply src/templates/plugin.routing.js:91]: It's ending up as `/`

File: src/helpers/routes.js
Comment (divine) [reply src/helpers/routes.js]: Good catch, I will add a parenthesis!

File: test/browser.test.js
Comment (divine) [reply test/browser.test.js]: Sorry, there is no problem with the static version, only on the server version.

I will update the tests ASAP.

Thanks!

**Comment by divine** [comment]
Good to go, let me know if anything is left @rchl 

Thanks!

File: yarn.lock
Comment (rchl) [inline yarn.lock:12221]: Interesting...
This might end up being an issue because importing `ufo` from plugin files might pick up the version used by Nuxt, which doesn't have the fix. Unless we do a trick like we do for `klona` where we create an alternative alias...

Comment (divine) [reply yarn.lock:12221]: I hope it's what you've asked for.

Thanks!

Comment (divine) [reply yarn.lock:12221]: Aliasing fails, I don't have much knowledge on how to fix those 🤷

Any help @rchl?

Thanks!

Comment (rchl) [reply yarn.lock:12221]: The problem is that `utils-common.js` is used both in the webpack context and in the pure module context where alias is not available.

Comment (divine) [reply yarn.lock:12221]: What can be done?

Thanks!

Comment (rchl) [reply yarn.lock:12221]: I'll try to move stuff around.

**Review by rchl — APPROVED** [review]

File: src/helpers/routes.js
Comment (divine) [inline src/helpers/routes.js:1]: This should `'~i18n-ufo'` @rchl?

Thanks!

File: test/module.test.js
Comment (divine) [inline test/module.test.js:5]: And this as well?

**Review by divine — COMMENTED** [review]
Looks like we've again `'ufo'` instead of `'~i18n-ufo'`

File: test/module.test.js
Comment (rchl) [reply test/module.test.js:5]: This is OK in the tests and in the context of the module since then the normal node importing mechanism will ensure that the project-specific version of ufo is picked up.

It's only a problem for webpack context - the files that are added through `addTemplate` or `addPlugin`.

Comment (divine) [reply test/module.test.js:5]: @rchl thank you very much for making it clear!

**Pull request merged**
